src/binary/ibc-setup/commands/init.ts

```diff
@@ -0,0 +1,58 @@
+import fs from 'fs';
+import path from 'path';
+
+import { Command } from 'commander';
+
+import { GlobalOptions } from '../types';
+
+type Options = GlobalOptions & {
+  src: string;
+  dest: string;
+};
+
+type ProgramOptions = Partial<Options>;
+
+function resolveHome(home?: string) {
+  if (home) {
+    return home;
+  }
+
+  if (!process.env.HOME) {
+    throw new Error('$HOME environment variable is not set.');
+  }
+
+  return `${process.env.HOME}/.ibc-setup`;
+}
+
+const REGISTRY_FILE = 'registry.yaml';
+
+export function init(_: unknown, program: Command) {
+  const programOptions: ProgramOptions = program.opts();
+
+  // TODO: Resolve options with precedence (env, command line etc.)
+  const options = {
+    src: programOptions.src ?? 'src',
+    dest: programOptions.dest ?? 'dest',
+    home: resolveHome(programOptions.home),
+  };
+
+  run(options);
+}
+
+function run(options: Options) {
+  if (!fs.existsSync(options.home)) {
+    fs.mkdirSync(options.home, { recursive: true });
+    console.log(`Initialized home directory at ${options.home}`);
+  } else if (!fs.lstatSync(options.home).isDirectory()) {
+    throw new Error(`${options.home} must be a directory. It is a file.`);
+  }
+
+  const REGISTRY_FILE_PATH = path.join(options.home, REGISTRY_FILE);
+  if (!fs.existsSync(REGISTRY_FILE_PATH)) {
+    // TODO: download registry.yaml from default location
+  } else if (!fs.lstatSync(REGISTRY_FILE_PATH).isFile()) {
+    throw new Error(`${REGISTRY_FILE_PATH} must be a file. It is a directory.`);
+  }
+
+  // TODO: Read registry.yaml
```

Comment:
I think this should be it's own function (when implemented), it will be used by almost every command.
The setup above (create if needed, etc) is great to have just in this `run` function.

I like the separation of external `init` and internal `run` (but maybe we will need to export `run` to cover it with tests?)

Reply:
Writing that function and testing it will close #51

The rest of this closes #55

Reply:
The `run` will be tested for sure so it must be exported.

Reply:
>I think this should be it's own function (when implemented), it will be used by almost every command.

I guess it's going to be reused but for now I just had to put it somewhere. Refactors is ahead!